src/extension/webView.ts

```diff
@@ -107,28 +107,35 @@ export async function createRepository(context: vscode.ExtensionContext) {
           }
 
           // Go to target folder & create app folder
-          const createAppFolder = `mkdir -p '${options.targetFolder}${options.name}' && cd '${options.targetFolder}${options.name}'`;
-          const openProject = `npm --no-git-tag-version version '${options.version}' && code .`;
-          const installDependencies = options.nodeModules ? '--install' : '--no-install';
-          const npmVersion = parseInt(execSync('npm --version').toString('utf8').split('')[0]);
-          const isLegacyNpm = npmVersion >= 7;
-          const sep = isLegacyNpm ? '' : '--';
-
-          if (options.repoType === 'piral') {
-            // Handle Piral Instance
-            const scaffoldPiral = `npm init piral-instance ${sep}registry '${options.npmRegistry}' ${sep}bundler '${options.bundler}' ${sep}defaults ${installDependencies}`;
-            runCommand(`${createAppFolder} && ${scaffoldPiral} && ${openProject}`);
-
-            // Dispose Webview
-            disposeWebview();
-          } else if (options.repoType === 'pilet') {
-            // Handle Pilet Instance
-            const scaffoldPilet = `npm init pilet ${sep}source '${options.piralPackage}' ${sep}registry '${options.npmRegistry}' ${sep}bundler '${options.bundler}' ${sep}defaults ${installDependencies}`;
-            runCommand(`${createAppFolder} && ${scaffoldPilet} && ${openProject}`);
-
-            // Dispose Webview
-            disposeWebview();
-          }
+          exec('npm --version', (error, stdout) => {
```

Comment:
There is really no need to change the whole code.

Put the logic for determining the npm version in a different module. Have two functions in there:

1. `async getNpmVersion(): string` (gets the full version string, e.g., `1.2.3`)
2. `async isLegacyNpmVersion(): boolean`, which uses `getNpmVersion`, gets only the major part of the version and then decides if its below 7 (that is a legacy version).

In this code call `await isLegacyNpmVersion()` and then continue with the result of the evaluation.